CommonWallet/Classes/Modules/AccountList/AccountModuleViewModelFactory.swift

```diff
@@ -24,51 +24,144 @@ final class AccountModuleViewModelFactory {
+        viewModel.delegate = delegate
+        return viewModel
     }
 }
 
 extension AccountModuleViewModelFactory: AccountModuleViewModelFactoryProtocol {
     func createViewModel(from balances: [BalanceData],
                          delegate: ShowMoreViewModelDelegate?) throws -> AccountModuleViewModel {
-        var viewModels = try context.viewModelFactories.map { try $0() }
+
+        var viewModels = try context.viewModelFactoryContainer.viewModelFactories.map { try $0() }
 
         var collapsingRange = viewModels.count..<viewModels.count
 
         if balances.count > 0 {
-            let assetViewModels: [AssetViewModelProtocol] = try balances.compactMap { (balance) in
+            let assetViewModels: [AssetViewModelProtocol] = balances.compactMap { (balance) in
                 guard let asset = assets.first(where: { $0.identifier.identifier() == balance.identifier }) else {
                     return nil
                 }
 
-                return try context.assetViewModelFactory(asset, balance, commandFactory)
+                if let assetViewModel = context.accountListViewModelFactory?
+                    .createAssetViewModel(for: asset, balance: balance, commandFactory: commandFactory) {
+                    return assetViewModel
+                } else {
+                    return createDefaultAssetViewModel(for: asset, balance: balance)
+                }
             }
 
             var assetsBlockLength = assetViewModels.count
 
-            let upper = context.assetsIndex + assetsBlockLength
-            let lower = context.assetsIndex + Int(context.minimumVisibleAssets)
+            let upper = context.viewModelFactoryContainer.assetsIndex + assetsBlockLength
+            let lower = context.viewModelFactoryContainer.assetsIndex + Int(context.minimumVisibleAssets)
             collapsingRange = min(lower, upper)..<upper
 
-            viewModels.insert(contentsOf: assetViewModels, at: context.assetsIndex)
+            viewModels.insert(contentsOf: assetViewModels, at: context.viewModelFactoryContainer.assetsIndex)
 
             if !collapsingRange.isEmpty {
-                let showMoreViewModel = try context.showMoreViewModelFactory(delegate)
-                viewModels.append(showMoreViewModel)
+                if let showMoreViewModel = context.accountListViewModelFactory?.createShowMoreViewModel(for: delegate) {
+                    viewModels.append(showMoreViewModel)
+                } else {
+                    let showMoreViewModel = createDefaultShowMoreViewModel(with: delegate)
+                    viewModels.append(showMoreViewModel)
+                }
 
                 assetsBlockLength += 1
             }
 
-            let actionsIndex = context.actionsIndex + assetsBlockLength - 1
+            let assetId: IRAssetId?
+
+            if assets.count == 1 {
+                assetId = assets.first?.identifier
+            } else {
+                assetId = nil
```

Comment:
Why nil? What if assets > 1?

Reply:
As you can see here we are creating view model for actions and we want to be sure that if there is only a single asset then, for example, send action will pass it further without choosing from the whole list of possible assets. This logic is necessary, for example, on account details screen.